### PR TITLE
create restful api http server

### PR DESCRIPTION
**Pull request by vitalis-virtus**

### CODE REVIEW

**Comment by penkova** [comment]
Please add `.idea` to gitignore

**Comment by penkova** [comment]
add `main.exe` to gitignore too

File: internal/app/store/store.go [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+import (
+	"database/sql"
+	"fmt"
+	_ "github.com/go-sql-driver/mysql"
+)
+
+type Store struct {
+	config *Config
+	Db     *sql.DB
```
Comment (penkova) [inline internal/app/store/store.go]: rename it to `DB`

File: internal/app/store/config.go [hunk 2]
```diff
@@ -0,0 +1,9 @@
+package store
+
+type Config struct {
+	DatabaseUrl string `toml:"database_url"`
```
Comment (penkova) [inline internal/app/store/config.go]: rename it to `DatabaseURL`

File: internal/app/apiserver/apiserver_internal_test.go [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+)
+
+func TestAPIServer_HandleHello(t *testing.T) {
+	s := New(NewConfig())
+
+	rec := httptest.NewRecorder()
+
+	req, _ := http.NewRequest(http.MethodGet, "/hello", nil)
+
+	s.handleHello().ServeHTTP(rec, req)
```
Comment (penkova) [inline internal/app/apiserver/apiserver_internal_test.go:17]: seems this part doesn't work

File: internal/app/apiserver/apiserver.go [hunk 4]
```diff
@@ -0,0 +1,244 @@
...
+		defer res.Close()
+
+		for res.Next() {
+			err := res.Scan(&author.ID, &author.Name, &author.Surname)
+			if err != nil {
+				log.Fatal(err)
+			}
+		}
+
+		resD, err := s.store.Db.Query(fmt.Sprintf("DELETE FROM booksdb.authors WHERE id=%v", authorId))
```
Comment (penkova) [inline internal/app/apiserver/apiserver.go]: need to check err here

File: internal/app/apiserver/apiserver.go [hunk 5]
```diff
@@ -0,0 +1,244 @@
...
+
+}
+
+func (s *APIServer) GetAuthor() http.HandlerFunc {
+	// here we can initialize variables only for this handler
+	var author model.Author
+
+	return func(w http.ResponseWriter, r *http.Request) {
+		vars := mux.Vars(r)
+		authorId := vars["id"]
```
Comment (penkova) [inline internal/app/apiserver/apiserver.go]: > `authorID`

File: internal/app/apiserver/apiserver.go [hunk 6]
```diff
@@ -0,0 +1,244 @@
...
+
+}
+
+func (s *APIServer) UpdateAuthor() http.HandlerFunc {
+	// here we can initialize variables only for this handler
+	var name, surname string
+
+	return func(w http.ResponseWriter, r *http.Request) {
+		vars := mux.Vars(r)
+		authorId := vars["id"]
```
Comment (penkova) [inline internal/app/apiserver/apiserver.go]: > `authorID`

File: internal/app/apiserver/apiserver.go [hunk 7]
```diff
@@ -0,0 +1,244 @@
...
+
+}
+
+func (s *APIServer) DeleteAuthor() http.HandlerFunc {
+	// here we can initialize variables only for this handler
+	var author model.Author
+
+	return func(w http.ResponseWriter, r *http.Request) {
+		vars := mux.Vars(r)
+		authorId := vars["id"]
```
Comment (penkova) [inline internal/app/apiserver/apiserver.go]: > `authorID`

File: internal/app/apiserver/apiserver.go [hunk 8]
```diff
@@ -0,0 +1,244 @@
...
+	return nil
+}
+
+// handlers ...
+func (s *APIServer) GetAuthors() http.HandlerFunc {
+	// here we can initialize variables only for this handler
+	var authors []model.Author
+
+	return func(w http.ResponseWriter, r *http.Request) {
+		res, err := s.store.Db.Query(fmt.Sprintf("SELECT * FROM booksdb.authors"))
```
Comment (penkova) [inline internal/app/apiserver/apiserver.go]: unnecessary use of fmt.Sprintf here

File: internal/app/apiserver/apiserver.go [hunk 9]
```diff
@@ -0,0 +1,244 @@
...
+	if err := st.Open(); err != nil {
+		return err
+	}
+
+	s.store = st
+
+	return nil
+}
+
+// handlers ...
```
Comment (penkova) [inline internal/app/apiserver/apiserver.go]: bad comment

File: internal/app/store/store.go (hunk 1 above)
Comment (vitalis-virtus) [reply internal/app/store/store.go]: renamed

File: internal/app/store/config.go (hunk 2 above)
Comment (vitalis-virtus) [reply internal/app/store/config.go]: renamed

File: internal/app/apiserver/apiserver_internal_test.go (hunk 3 above)
Comment (vitalis-virtus) [reply internal/app/apiserver/apiserver_internal_test.go:17]: fixed it

File: internal/app/apiserver/apiserver.go (hunk 4 above)
Comment (vitalis-virtus) [reply internal/app/apiserver/apiserver.go]: add error check

File: internal/app/apiserver/apiserver.go (hunk 5 above)
Comment (vitalis-virtus) [reply internal/app/apiserver/apiserver.go]: renamed

File: internal/app/apiserver/apiserver.go (hunk 6 above)
Comment (vitalis-virtus) [reply internal/app/apiserver/apiserver.go]: renamed

File: internal/app/apiserver/apiserver.go (hunk 7 above)
Comment (vitalis-virtus) [reply internal/app/apiserver/apiserver.go]: renamed

File: internal/app/apiserver/apiserver.go (hunk 8 above)
Comment (vitalis-virtus) [reply internal/app/apiserver/apiserver.go]: deleted fmt.Sprintf

File: internal/app/apiserver/apiserver.go (hunk 9 above)
Comment (vitalis-virtus) [reply internal/app/apiserver/apiserver.go]: add good comments to each handle func

**Comment by vitalis-virtus** [comment]
Added gitignore, readme, edited Makefile

**Review by penkova — APPROVED** [review]

**Pull request merged**